Compile bug: mtmd-cli fails to compile with `nix build`

### Git commit

cb8f4fa3f8e359e61ed570adb8a3c2ce74be5179 and after doesn't compile.

54910bd4f3762b716e90b658beb68d349571057e does compile.

### Operating systems

Mac

### GGML backends

Metal

### Problem description & steps to reproduce

Running `nix build` on Mac (M1) after tag b8195 produces an error compiling mtmd-cli. Error pasted in the log section.

I checked out 54910bd4f3762b716e90b658beb68d349571057e (the commit before) and that compiles fine with `nix build`.

### First Bad Commit

cb8f4fa3f8e359e61ed570adb8a3c2ce74be5179 appears to be the first bad commit.

### Compile command

```shell
nix build
```

### Relevant log output

```shell
> [364/447] Building CXX object tools/mtmd/CMakeFiles/llama-mtmd-cli.dir/mtmd-cli.cpp.o
       > FAILED: tools/mtmd/CMakeFiles/llama-mtmd-cli.dir/mtmd-cli.cpp.o
       > /nix/store/v39c0h6xv6hvki2k1bsyp5n090q8y2bp-clang-wrapper-16.0.6/bin/clang++ -DGGML_BACKEND_SHARED -DGGML_SHARED -DGGML_USE_CPU -DGGML_USE_METAL -DLLAMA_SHARED -I/tmp/nix-build-llama-cpp-metalkit-0.0.0.drv-0/source/common/. -I/tmp/nix-build-llama-cpp-metalkit-0.0.0.drv-0/source/common/../vendor -I/tmp/nix-build-llama-cpp-metalkit-0.0.0.drv-0/source/src/../include -I/tmp/nix-build-llama-cpp-metalkit-0.0.0.drv-0/source/ggml/src/../include -I/tmp/nix-build-llama-cpp-metalkit-0.0.0.drv-0/source/tools/mtmd/. -O3 -DNDEBUG -isysroot /nix/store/amdympl4rz7kj93j82cva60v8007n4nv-apple-sdk-11.3/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -Wmissing-declarations -Wmissing-noreturn -Wall -Wextra -Wpedantic -Wcast-qual -Wno-unused-function -Wunreachable-code-break -Wunreachable-code-return -Wmissing-prototypes -Wextra-semi -MD -MT tools/mtmd/CMakeFiles/llama-mtmd-cli.dir/mtmd-cli.cpp.o -MF tools/mtmd/CMakeFiles/llama-mtmd-cli.dir/mtmd-cli.cpp.o.d -o tools/mtmd/CMakeFiles/llama-mtmd-cli.dir/mtmd-cli.cpp.o -c /tmp/nix-build-llama-cpp-metalkit-0.0.0.drv-0/source/tools/mtmd/mtmd-cli.cpp
       > /tmp/nix-build-llama-cpp-metalkit-0.0.0.drv-0/source/tools/mtmd/mtmd-cli.cpp:277:5: error: no member named 'setlocale' in namespace 'std'; did you mean simply 'setlocale'?
       >     std::setlocale(LC_NUMERIC, "C");
       >     ^~~~~~~~~~~~~~
       >     setlocale
       > /nix/store/amdympl4rz7kj93j82cva60v8007n4nv-apple-sdk-11.3/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/locale.h:53:8: note: 'setlocale' declared here
       > char            *setlocale(int, const char *);
       >                  ^
       > 1 error generated.
```

--- a/tools/mtmd/mtmd-cli.cpp
+++ b/tools/mtmd/mtmd-cli.cpp
@@ -13,6 +13,7 @@
 #include <vector>
 #include <limits.h>
 #include <cinttypes>
+#include <clocale>
 
 #if defined (__unix__) || (defined (__APPLE__) && defined (__MACH__))
 #include <signal.h>

